### ISSUE TITLE
[Bug] Can not save NULL value for relation field

### ISSUE THREAD

**Issue by mialex** [Laravel-Backpack/CRUD#3646]
# Bug report

### What I did

I have two models `\App\Models\Product` and `\App\Models\Settings`.
The model `\App\Models\Product` has one-to-one relation with settings:
```
    /**
     * @return \Illuminate\Database\Eloquent\Relations\HasOne
     */
    public function settings()
    {
        return $this->hasOne(
            \App\Models\Settings::class, 'product_id', 'id'
        );
    }
```

I add a **select2_from_array** field to be able to choose a discount option and remove the option value.

```
$this->crud->addField(
            [
                'label'       => 'Product discount',
                'type'        => 'select2_from_array',
                'name'        => 'settings.discount_type_id',
                'attribute'   => 'discount_type_id',
                'allows_null' => true,
                'options'     => [
                    1 => '5 %',
                    2 => '10 %',
                    3 => '15 %',
                ],
                
            ]
        );
```

### What I expected to happen

I expected to choose an option and have it key saved in `\App\Models\Settings` and when I choose the empty option (as I have the field option `'allows_null' => true,`) I expect this value to be set to `null`

### What happened

When I choose one of the values from the list, it's saved to the relation, but when I choose an empty option - it's not saved.

As I can see the code here:
https://github.com/Laravel-Backpack/CRUD/blob/master/src/app/Library/CrudPanel/Traits/Create.php#L222

`! is_null(Arr::get($data, $attributeKey))`

If the relation value is `null` then it's not possible to store it.

Are there any solution to solve that? 

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

```
### PHP VERSION:
PHP 7.2.34 (cli) (built: Mar 16 2021 12:25:22) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v3.0.2, Copyright (c) 2002-2021, by Derick Rethans
    with Zend OPcache v7.2.34, Copyright (c) 1999-2018, by Zend Technologies

### LARAVEL VERSION:
v7.29.2@0fb15ae253c3a043f2c5cbf9641427a7646ea77a

### BACKPACK VERSION:
4.1.26@ae68ca24844929ac38d2e792197551e823c43570
```

**Comment by mialex** [comment]
Any new or feedback, details about the issue?

**Comment by pxpm** [comment]
Hello @mialex 

Sorry for taking time to come back here. 

It is indeed a bug. I'v fixed it already in #3410 

![image](https://user-images.githubusercontent.com/7188159/117441761-c77a4400-af2d-11eb-9771-a30abdc7f406.png)

I went ahead and submitted #3705 because I think it will take a little bit more time to merge #3410 

I am going to close this since there is already the PR open, and we keep any discussion regarding this issue in the PR thread.

Best,
Pedro
